docker/dev/container/common/locking.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Try to close {lockfd} and remove lockfile.
+#
+# this function shall be called only after an `flock` command
+#  has obtained a lock on the named file descriptor {lockfd}.
+#
+# {lockfd}: named file descriptor
+# $1: lockfd
+# $2: lockfile
+function clean_lock() {
```

Comment:
The lockfile has to reside in a directory which is shared by all the users of the shared directory (volume) `/opt/pycroft/venv`. Since different containers access this volume, they have potentially different file systems. Using a file that resides there seems like a failsafe approach.

Removing the file may add complications, but has the advantage that a leftoverfile makes bugs visible.
I am not sure which option is best.

Reply:
You could try to use the `/opt/pycroft/venv` directory itself.

Reply:
(the mail replying feature did not work properly)

> That would be possible and I think a more elegant solution. However, one would have to ensure the directory exists beforehand (The venv directory should always exist, but the node_modules not necessarily).

I removed the complicated cleanup logic in favor of the lokdirs `venv/` and `app/node_modules`.